object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/IndexedFieldProvider.java

```diff
@@ -62,47 +62,49 @@ interface FieldIndexingMetadata {
       boolean isVector(String[] propertyPath);
 
       Object getNullMarker(String[] propertyPath);
-   }
-
-   FieldIndexingMetadata NO_INDEXING = new FieldIndexingMetadata() {
-      @Override
-      public boolean isSearchable(String[] propertyPath) {
-         return false;
-      }
-
-      @Override
-      public boolean isAnalyzed(String[] propertyPath) {
-         return false;
-      }
-
-      @Override
-      public boolean isNormalized(String[] propertyPath) {
-         return false;
-      }
 
-      @Override
-      public boolean isProjectable(String[] propertyPath) {
-         return false;
-      }
+      TypeMetadata keyType(String property);
 
-      @Override
-      public boolean isAggregable(String[] propertyPath) {
-         return false;
-      }
-
-      @Override
-      public boolean isSortable(String[] propertyPath) {
-         return false;
-      }
-
-      @Override
-      public boolean isVector(String[] propertyPath) {
-         return false;
-      }
+   }
 
-      @Override
-      public Object getNullMarker(String[] propertyPath) {
-         return null;
-      }
-   };
+   static <TM> FieldIndexingMetadata<TM> noIndexing() {
```

Comment:
Invoking this method will now instantiate a new `FieldIndexingMetadata` every time. Previously it was reusing the same static instance

Reply:
Right! I'm going to make also the instance static.